### PR TITLE
Support shoots/adminkubeconfig

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
The host and target credential can now be referenced by pointing to the Shoot (`spec.host.credentials.shootRef`, `spec.target.credentials.shootRef`). In this case the an `AdminKubeconfigRequest` will be `created`. It is validated that the user has the permission to do so.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
https://github.com/gardener/dashboard/issues/1167

**Release note**:

```feature developer
In case of a `Shoot` cluster, you can now reference it with `spec.host.credentials.shootRef` and `spec.target.credentials.shootRef` instead of referencing the kubeconfig secret.
```

### CODE REVIEW

**Review by grolu — APPROVED** [review]
/lgtm

File: api/v1alpha1/terminal_types.go [hunk 1]
```diff
@@ -254,13 +254,25 @@ type Container struct {
...
 	// +optional
 	ServiceAccountRef *corev1.ObjectReference `json:"serviceAccountRef,omitempty"`
+
+	// ShootRef references the shoot cluster. The admin kubeconfig retrieved from the shoots/adminkubeconfig endpoint is used
+	// +optional
+	ShootRef *ShootRef `json:"shootRef,omitempty"`
+}
+
+// ShootRef references the shoot cluster by namespace and name
+type ShootRef struct {
```
Comment (holgerkoser) [inline api/v1alpha1/terminal_types.go:271]: I think we already discussed it somewhere. Why don't we use `NamespacedName` https://github.com/kubernetes/apimachinery/blob/release-1.24/pkg/types/namespacedname.go here. And the json annotations could be omitted.

File: controllers/terminal_controller.go [hunk 2]
```diff
@@ -1327,6 +1335,43 @@ func NewClientSetFromServiceAccountRef(ctx context.Context, cs *ClientSet, ref *
 	})
 }
 
+func NewClientSetFromShootRef(ctx context.Context, cs *ClientSet, ref *extensionsv1alpha1.ShootRef, scheme *runtime.Scheme) (*ClientSet, error) {
```
Comment (holgerkoser) [inline controllers/terminal_controller.go:1338]: Why not move all the code related to client creation into an own package and globally initialize the scheme there.

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Pull request merged**